pkg/provisioner/ironic/ironic.go

```diff
@@ -1743,42 +1743,67 @@ func (p *ironicProvisioner) PowerOn() (result provisioner.Result, err error) {
 
 	if ironicNode.PowerState != powerOn {
 		if ironicNode.TargetPowerState == powerOn {
-			p.log.Info("waiting for power status to change")
-			return operationContinuing(powerRequeueDelay)
+			if ironicNode.LastError != "" {
```

Comment:
Is it "" or nil? From API perspective it's JSON null by default.

Reply:
I believe it is "". Also, used here https://github.com/metal3-io/baremetal-operator/blob/master/pkg/provisioner/ironic/ironic.go#L585